CMakeLists.txt

```diff
@@ -1,7 +1,10 @@
 project(charon NONE)
 cmake_minimum_required(VERSION 3.0.2) #Tested only with 3.6.1 and 3.9.1.
 
-find_package(PythonInterp 3.4.2 REQUIRED)
+# Use FindPython 3 because if we use 3.4 for instance, it will prefer 3.4, even if there is another
+# Python (say 3.5) in CMAKE_PREFIX_PATH which also meets the requirement. We prefer the Python in
+# CMAKE_PREFIX_PATH if set for the Cura building process, so we specify Python 3 here.
```

Comment:
So far we've always let cura-build-environment decide on the exact version and did a smoke test to see if the build breaks when updating the environment.

I think that workflow is a requirement now because this library is used by Cura as well as the UM3 and UMS5 firmware. If we specify an exact version of CPython here, we're locking all of our software products to be using the same CPython version for eternity unless they all upgrade at the same time.

If we don't lock it though, we run the risk that CPython breaks its backward compatibility. In general though we should be able to assume that this doesn't happen or at least not very often within the 3.x releases.

Reply:
That's a 'no' then :-)